study.md

```diff
@@ -21,18 +21,29 @@ There is no requirement to follow any links from these pages.
 After doing all the required readings please tell us anything you were confused about.
 
 ```md
-<!-- your answer here -->
+I think the rules of return statements are something I still struggle with. My understanding was that as soon as you see a return statement in any context, the function essentially stops running and returns whatever it is told to return at that moment, but clearly that is not the case everytime. I will read this article over a few more times.
```

Comment:
```return``` will return control flow to the function that called it.  Here's an example that I wrote.  Let me know if it's unclear.
```javascript
function a() {
  console.log('a(): Inside of a().');
  console.log('a(): Calling b() and setting its return value to bValue.');

  const bValue = b();

  console.log('a(): b() has returned.  Control has been given back to a().');
  console.log('a(): Printing bValue: ', bValue);
  console.log('a(): Returning a\'s value.');

  return 'It\'s me, a.';
}

function b() {
  console.log('b(): Inside of b().');
  console.log('b(): Calling c() and setting its return value to cValue.');

  const cValue = c();

  console.log('b(): c() has returned.  Control has been given back to b()');
  console.log('b(): Printing cValue:', cValue);
  console.log('b(): Returning B\'s value.');

  return 'It\'s me, b.';
}

function c() {
  console.log('c(): Inside of c().');
  console.log('c(): Returning c\'s value.');

  return 'It\'s me, c.';
}

console.log(
  'global: Calling a() from the global scope and setting its value to aValue.'
);

const aValue = a();

console.log(
  'global: a() has returned.  Control has been given back to the global scope.'
);
console.log('global: Printing aValue:', aValue);
console.log('global: Script finished.');

// Output:
// global: Calling a() from the global scope and setting its value to aValue.
// a(): Inside of a().
// a(): Calling b() and setting its return value to bValue.
// b(): Inside of b().
// b(): Calling c() and setting its return value to cValue.
// c(): Inside of c().
// c(): Returning c's value.
// b(): c() has returned.  Control has been given back to b()
// b(): Printing cValue: It's me, c.
// b(): Returning B's value.
// a(): b() has returned.  Control has been given back to a().
// a(): Printing bValue:  It's me, b.
// a(): Returning a's value.
// global: a() has returned.  Control has been given back to the global scope.
// global: Printing aValue: It's me, a.
// global: Script finished.
```